src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,437 @@
+#include "encoder/encoderfdkaac.h"
+
+#ifdef __APPLE__
+#include <QCoreApplication>
+#endif
+#include <QDir>
+#include <QStandardPaths>
+#include <QString>
+#include <QStringList>
+
+#include "engine/sidechain/enginesidechain.h"
+#include "recording/defs_recording.h"
+#include "util/logger.h"
+#include "util/sample.h"
+
+namespace {
+// recommended in encoder documentation, section 2.4.1
+const int kOutBufferBits = 6144;
+const mixxx::Logger kLogger("EncoderFdkAac");
+} // namespace
+
+EncoderFdkAac::EncoderFdkAac(EncoderCallback* pCallback)
+        : aacEncOpen(nullptr),
+          aacEncClose(nullptr),
+          aacEncEncode(nullptr),
+          aacEncInfo(nullptr),
+          aacEncoder_SetParam(nullptr),
+          m_aacAot(AOT_AAC_LC),
+          m_bitrate(0),
+          m_channels(0),
+          m_samplerate(0),
+          m_pCallback(pCallback),
+          m_pLibrary(nullptr),
+          m_pInputFifo(nullptr),
+          m_pFifoChunkBuffer(nullptr),
+          m_readRequired(0),
+          m_aacEnc(),
+          m_pAacDataBuffer(nullptr),
+          m_aacInfo(),
+          m_hasSbr(false) {
+    // Load the shared library
+    //
+    // Libraries from external sources take priority because they may include HE-AAC support,
+    // but we do not risk shipping that with Mixxx because of patents and GPL compatibility.
+    //
+    // From https://bugzilla.redhat.com/show_bug.cgi?id=1501522#c112 :
+    // The Fedora Project is aware that the Free Software Foundation
+    // has stated that the Fraunhofer FDK AAC license is GPL
+    // incompatible, specifically, because of Clause 3.
+    //
+    // We believe that the fdk-aac software codec implementation that we
+    // wish to include in Fedora [which is shipped with Mixxx on Windows and macOS]
+    // is no longer encumbered by AAC patents.
+    // This fact means that Clause 3 in the FDK AAC license is a "no op",
+    // or to put it plainly, if no patents are in play, there are no
+    // patent licenses to disclaim. For this (and only this) specific
+    // implementation of fdk-aac, we believe that the FDK AAC license is
+    // GPL compatible.
+    QStringList libnames;
+#if __WINDOWS__
+    // Search for library from B.U.T.T.
+    QString buttFdkAacPath = buttWindowsFdkAac();
+    if (!buttFdkAacPath.isEmpty()) {
+        kLogger.debug() << "Found libfdk-aac at" << buttFdkAacPath;
+        libnames << buttFdkAacPath;
+    }
+#elif __APPLE__
+    // Homebrew
+    libnames << QStringLiteral("/usr/local/lib/libfdk-aac");
+    // MacPorts
+    libnames << QStringLiteral("/opt/local/lib/libfdk-aac");
+
+    // Mixxx application bundle
+    QFileInfo bundlePath(QCoreApplication::applicationDirPath() +
+            QStringLiteral("/../Frameworks/libfdk-aac"));
+    libnames << bundlePath.absoluteFilePath();
+#endif
+    libnames << QStringLiteral("fdk-aac");
```

Comment:
This code does not work for me. It obfuscates which library is really loaded, Because it relies on target specific code here and in the Qt library. Only if you know both you understand what happen. 
The solution in https://github.com/mixxxdj/mixxx/pull/3543 works much better, readable like a book:
https://github.com/mixxxdj/mixxx/blob/b660eb23165bf04b30f2d9c133758c35f38d2b90/src/encoder/encoderfdkaac.cpp#L66

Reply:
Can we please stop quibbling over this inconsequential implementation detail and get a release out?

Reply:
I still think reimplementing abstractions that Qt provides for us is a bad idea.

Reply:
The Qt abstraction is made for the case Qt considers as the regular case. But we are not at the regular case here and we need to hack around it. The proposed code is only readable if you full understand the QT abstraction. I had to read the Qt source to fully understand it. In the original implementation we build a easy to understand string list, tweak-able per target.

And by the way, if this is so quibbling for you, just revert it and we can make progress.

Reply:
I have tested this on 3 OSes. It works. Qt documentation explains how it works. I am not putting any more effort into this laborious process of testing that everything works across 3 OSes with an alternative implementation reimplementing what Qt already does for us.

Reply:
The original version works as well. Now we have cluttered hard to maintain version that relies on Qt implementation details.